pkg/test/context.go

```diff
@@ -23,12 +23,20 @@ import (
 )
 
 type TestCtx struct {
-	ID         string
-	CleanUpFns []finalizerFn
-	Namespace  string
+	id         string
+	cleanupFns []cleanupFn
+	namespace  string
+	t          *testing.T
 }
 
-type finalizerFn func() error
+type CleanupOptions struct {
+	TestContext   *TestCtx
+	Timeout       time.Duration
+	RetryInterval time.Duration
+	SkipPolling   bool
```

Comment:
Let's remove `SkipPolling`. And we can skip polling if both Timeout and RetryInterval are set to 0.